plugins/modules/ndo_interface_setting.py

```diff
@@ -0,0 +1,808 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2024, Anvitha Jain (@anvjain) <anvjain@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_interface_setting
```

Comment:
In the l3out module Sabari introduced dictionary like input for protocol specific configuration and also for l3out. Should we do the same here for MCP, LLDP, Link Level etc...?

See PR: https://github.com/CiscoDevNet/ansible-mso/pull/519/files

Reply:
Changed only for MCP and LLDP to dict, as enabling or disabling cretain parameters affects the others.